models/user.modal.go

```diff
@@ -0,0 +1,83 @@
+package models
+
+import (
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+)
+
+type User struct {
+	ID        *uuid.UUID `gorm:"type:uuid;default:uuid_generate_v4();primary_key"`
+	FirstName string     `gorm:"type:varchar(100);not null"`
+	LastName  string     `gorm:"type:varchar(100);not null"`
+	Username  string     `gorm:"type:varchar(100);uniqueIndex;not null"`
+	Role      *string    `gorm:"type:varchar(50);default:'user';not null"`
+	Email     string     `gorm:"type:varchar(100);uniqueIndex;not null"`
+	Password  string     `gorm:"type:varchar(100);not null"`
+	Photo     *string    `gorm:"not null;default:'default.png'"`
+	CreatedAt *time.Time `gorm:"not null;default:now()"`
+	UpdatedAt *time.Time `gorm:"not null;default:now()"`
+}
+
+type SignUpInput struct {
+	FirstName       string `json:"first_name" validate:"required"`
+	LastName        string `json:"last_name" validate:"required"`
+	Username        string `json:"username" validate:"required"`
+	Email           string `json:"email" validate:"required"`
+	Password        string `json:"password" validate:"required,min=8"`
+	PasswordConfirm string `json:"password_confirm" validate:"required,min=8"`
+	Photo           string `json:"photo"`
+}
+
+type SignInInput struct {
+	Username string `json:"username" validate:"required"`
+	Password string `json:"password" validate:"required"`
+}
+
+type UserResponse struct {
+	ID        uuid.UUID `json:"id,omitempty"`
+	FirstName string    `json:"first_name,omitempty"`
+	LastName  string    `json:"last_name,omitempty"`
+	Username  string    `json:"username,omitempty"`
+	Email     string    `json:"email,omitempty"`
+	Photo     string    `json:"photo,omitempty"`
+	CreatedAt time.Time `json:"created_at"`
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+func FilterUserRecord(user *User) UserResponse {
+	return UserResponse{
+		ID:        *user.ID,
+		FirstName: user.FirstName,
+		LastName:  user.LastName,
+		Username:  user.Username,
+		Email:     user.Email,
+		Photo:     *user.Photo,
```

Comment:
do we need this here

Reply:
For return the response I have added filter user